prompterator/models/openai_models.py

```diff
@@ -86,12 +90,39 @@ def __init__(self):
         super().__init__()
 
     def call(self, idx, input, **kwargs):
+        structured_output_data: StructuredOutputData = kwargs["structured_output"]
         model_params = kwargs["model_params"]
         try:
```

Comment:
This try-except block has become massive. Let's make it as small as possible. Do you have in mind any exceptions that could happen within the newly added code blocks? If yes, let's perhaps catch them separately?

Reply:
I separated it into two functions